docs/tooling/cli/04-templates.md

```diff
@@ -0,0 +1,139 @@
+  - This setting will require that you have `moduleid` as one of the prompts.
+  - This setting takes higher precedence over `suffix`, so if you were to configure both, it would only apply the backend module prefix and ignore `suffix`.
+- `additionalActions` - There might be additional steps you want taken after your package is generated. See [Additional Actions](#additional-actions) for more details.
+
+Once you have your composed template yaml file, add your new template to the CLI config in your root `package.json`:
+
+```diff
+{
+  // ...
+  "backstage": {
+    "cli": {
+      // ...
++       "templates": [
++         {
++           "id": "my custom CLI template",
++           "target": "./custom-template.yaml"
++         }
++       ]
+    }
+  }
+}
+```
+
+If you'd like to see more examples, you can find all the default templates and its yaml files [here](https://github.com/backstage/backstage/tree/master/packages/cli/templates).
+
+### Prompts
+
+Prompts elicit values from users and replace variables in skeleton templates.
+
+There are four pre-defined prompts:
+
+```yaml
+prompts:
+  - id
+  - moduleid
+  - npmregistry
+  - owner
+```
+
+If your template requires more information, you can also configure your own custom prompts:
+
+```yaml
+prompts:
+  - id: color
+    prompt: Enter your favorite color
+    default: red
+    validate: backstage-id
+```
+
+- `id` - The `id` is the value that will be applied to your template. An `id` of `color`, like in the example above, will be used to replace placeholder values you configure in your skeleton template:
+  ```ts
+  // ./template/index.ts.hbs
+  console.log('My favorite color is {{color}}');
+  ```
+- `prompt` - The `prompt` value is the text that is displayed to the user:
+  ```
+  ? What do you want to create? custom-template
+  ? Enter the ID of the plugin [required] my-plugin
+  ? Enter your favorite color (red)
+  ```
+- `default` - You can also provide a `default` value for the convenience of the user.
+- `validate` - There is only one built-in validator at the moment and it is `backstage-id`. The `backstage-id` validator enforces the user-provided value to be all lowercase and only contain letters, digits, and dashes.
+
+### Additional Actions
+
+Additional actions are useful for running tasks. For example, if you want your newly created package to be automatically added as a dependency to `packages/backend/package.json` so that your users won't have to do so manually, you could add `install-backend` as one of the `additionalActions` to your template.
+
+There are four `additionalActions`:
+
+|        Action         | Description                                                            |
+| :-------------------: | :--------------------------------------------------------------------- |
+|  `install-frontend`   | Installs your new package/plugin as a dependency to `packages/app`     |
+| `add-frontend-legacy` | Adds your new frontend plugin to `packages/app/src/App.tsx`            |
```

Comment:
Do these actions need to be separate? Perhaps a single `add-to-frontend` action (and the same for backend) can do the whole task of integrating the plugin into the local app and backend packages.